geth/node/manager.go

```diff
@@ -518,3 +519,77 @@ func (m *NodeManager) isNodeAvailable() error {
 
 	return nil
 }
+
+// SyncAndStopNode waits until node synchronization and exits.
+// Reports errors to returned channel.
+func (m *NodeManager) SyncAndStopNode(timeout time.Duration) (errChan <-chan error) {
+	ch := make(chan error, 2)
+	errChan = ch
+	go m.syncAndStop(timeout, ch)
+	return
+}
+
+func (m *NodeManager) syncAndStop(timeout time.Duration, errChan chan<- error) {
+	// We need to have a larger timeout than ticker delta here.
+	if timeout < time.Second*2 {
+		errChan <- errors.New("Sync timeout is too short, must be at least two seconds")
+		m.stopAndReport(errChan)
+		return
+	}
+
+	// Don't wait for any blockchain sync for the
+	// local private chain as blocks are never mined.
+	if m.config.NetworkID == params.StatusChainNetworkID {
+		m.stopAndReport(errChan)
+		return
+	}
+
+	if m.lesService == nil {
+		errChan <- errors.New("LightEthereumService is nil")
+		m.stopAndReport(errChan)
+		return
+	}
+
+	timer := time.NewTimer(timeout)
+	defer timer.Stop()
+	ticker := time.NewTicker(1 * time.Second)
+	defer ticker.Stop()
+	for {
+		select {
+		case <-timer.C:
```

Comment:
I would use `<-time.After(timeout)` as it's used just once in contrast to `ticker`.

Reply:
:+1: